tracetools/include/tracetools/tp_call.h

```diff
@@ -299,6 +350,40 @@ TRACEPOINT_EVENT(
   )
 )
 
+TRACEPOINT_EVENT(
+  TRACEPOINT_PROVIDER,
+  rmw_send_request,
+  TP_ARGS(
+    const void *, rmw_client_handle_arg,
```

Comment:
it would be easier, but it would also be redundant. We can get the client GID from the `rmw_client_handle` thanks to the `rmw_client_init` tracepoint, which provides the mapping from `rmw_client_handle` to (client) `gid`. This is how the instrumentation works in general. This way, at runtime, we can (generally) minimize the size of the data we collect and simplify the tracepoint arguments, since we already have the `rmw_client_handle` and would need to fetch/copy the GID if we used that.

Does that make sense?

Reply:
yeah, it does. i think if it shows the client gid and send request / receive response information close together, that should be no problem. (i was thinking that this could be more consistent and would be easier to search for user if that prints the each message, but efficiency matters as you mentioned.) resolving, thanks for the explanation.